src/main/java/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -294,35 +298,33 @@ private static void updateMapUrlAndSizeLabel(final DownloadFileDescription map,
   }
 
   private static String createLabelText(final MapAction action, final DownloadFileDescription map) {
-    final String DOUBLE_SPACE = "&nbsp;&nbsp;";
+    final String doubleSpace = "&nbsp;&nbsp;";
 
-    final long mapSize;
-    String labelText = "<html>" + map.getMapName() + DOUBLE_SPACE + " v" + map.getVersion();
+    final StringBuilder sb = new StringBuilder();
+    sb.append("<html>" + map.getMapName() + doubleSpace + " v" + map.getVersion());
 
+    final Optional<Long> mapSize;
     if (action == MapAction.INSTALL) {
-      if (map.newURL() == null) {
-        mapSize = 0L;
-      } else {
-        mapSize = DownloadUtils.getDownloadLength(map.newURL()).orElse(-1);
-      }
+      final String mapUrl = map.getUrl();
+      mapSize = (mapUrl != null) ? DownloadUtils.getDownloadLength(mapUrl) : Optional.empty();
     } else {
-      mapSize = map.getInstallLocation().length();
-    }
-
-    if (mapSize > 0) {
-      labelText += DOUBLE_SPACE + " (" + createSizeLabel(mapSize) + ")";
+      mapSize = Optional.of(map.getInstallLocation().length());
     }
+    mapSize.ifPresent(size -> {
+      sb.append(doubleSpace + " (" + createSizeLabel(size) + ")");
+    });
 
-    labelText += "<br/>";
+    sb.append("<br/>");
```

Comment:
If we want to stick with html5 conventions/recommendations we can change this to just `<br>`